### PR TITLE
help page candidate completed

### PR DESCRIPTION
**Pull request by aha-wtag**
[JIRA](https://welltravel.atlassian.net/browse/RMS-15)
rms-15

### CODE REVIEW

File: app/assets/stylesheets/main.scss [hunk 1]
```diff
@@ -1083,3 +1083,30 @@ table tr td {
   display: none;
   padding: 6px 12px;
 }
+
+.help_link{
```
[the remark this answers is not in the document]
Comment (lra-wtag) [reply app/assets/stylesheets/main.scss]: please take care of this comment

[the remark this answers is not in the document]
Comment (lra-wtag) [reply app/assets/stylesheets/main.scss]: please take care of this comment

File: app/assets/stylesheets/main.scss [hunk 2]
```diff
@@ -1083,3 +1083,30 @@ table tr td {
...
+  width: 50px;
```
[the remark this answers is not in the document]
Comment (lra-wtag) [reply app/assets/stylesheets/main.scss]: please take care of this comment

File: app/assets/stylesheets/main.scss [hunk 3]
```diff
@@ -1083,3 +1083,30 @@ table tr td {
...
+  height: 50px;
+  float: right;
+  margin-right: 50px;
+  img {
```
[the remark this answers is not in the document]
Comment (lra-wtag) [reply app/assets/stylesheets/main.scss]: please take care of this comment

File: app/assets/stylesheets/main.scss [hunk 4]
```diff
@@ -1083,3 +1083,30 @@ table tr td {
...
+  }
+}
+
+.container {
+  .information-headline {
+    text-align: center;
+  }
+}
+
+.candidate.new {
```
[the remark this answers is not in the document]
Comment (lra-wtag) [reply app/assets/stylesheets/main.scss]: please take care of this comment

File: app/assets/stylesheets/main.scss [hunk 5]
```diff
@@ -1083,3 +1083,30 @@ table tr td {
...
+    .gender_issue
```
[the remark this answers is not in the document]
Comment (lra-wtag) [reply app/assets/stylesheets/main.scss]: please take care of this comment

[the remark this answers is not in the document]
Comment (lra-wtag) [reply app/assets/stylesheets/main.scss]: please take care of this comment

File: app/assets/stylesheets/main.scss [hunk 6]
```diff
@@ -1083,3 +1083,30 @@ table tr td {
...
+      label {
+        width: 8rem;
```
[the remark this answers is not in the document]
Comment (lra-wtag) [reply app/assets/stylesheets/main.scss]: please take care of this comment

File: app/controllers/pages_controller.rb [hunk 7]
```diff
@@ -10,4 +9,7 @@ def index
       redirect_to new_candidate_path
     end
   end
+
+  def help
```
[the remark this answers is not in the document]
Comment (lra-wtag) [reply app/controllers/pages_controller.rb]: please take care of this comment

File: app/views/candidates/new.html.erb [hunk 8]
```diff
@@ -1,6 +1,11 @@
 <div class ="candidate new">
   <br />
   <%= link_to t('shared.all.back'), pages_path %>
+
+  <div class="help_link">
+    <%= link_to image_tag('help_page_icon.png'), controller: :pages, action: :help %>
```
Comment (lra-wtag) [inline app/views/candidates/new.html.erb]: try removing the `controller: :pages, action: :help` and instead give the route

File: app/views/pages/help.html.erb [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+              <label for='step4' id="back-step4" class="back">
+                <div class="btn-back"><%= t('shared.all.prev') %></div>
+              </label>
+            </div>
+          </div>
+      </div>
+          <% end %>
+  </div>
+ </div>
+</div>
```
Comment (lra-wtag) [inline app/views/pages/help.html.erb]: add an empty line at the end of the code

File: config/locales/en.yml [hunk 10]
```diff
@@ -12,6 +12,26 @@ en:
...
+    help_page:
+      address: House# 111, Road# 9, Block# F, Bashundhara R/A
+      cgpa: 2.4
+      contact: "+8801XXXXXXXXX"
+      degree: Bachelor of Computer Science and Engineering
+      dob: 02/29/1992
+      email: johnsmith@yahoo.com
+      exp: 2
+      help: Information Help Page
+      help_info: This is help page to create your profile
```
Comment (lra-wtag) [inline config/locales/en.yml]: just `info` should be fine

File: config/locales/en.yml [hunk 11]
```diff
@@ -69,7 +89,7 @@ en:
       update: Update Job Info
     title: Job Title
     view: View Job Posts
-    view_desc: This link will show you all job posts (old and new)
+    view_desc: This link will help you all job posts (old and new)
```
Comment (lra-wtag) [inline config/locales/en.yml]: this isnt't suppose to be `help`, it should be`show`

File: config/locales/en.yml [hunk 12]
```diff
@@ -12,6 +12,26 @@ en:
     details: Candidate Details
     edit: Edit My Profile
     edu_profile: Educational Background
+    help_page:
+      address: House# 111, Road# 9, Block# F, Bashundhara R/A
+      cgpa: 2.4
```
Comment (lra-wtag) [inline config/locales/en.yml]: 2.40 is a better example

File: config/locales/en.yml [hunk 13]
```diff
@@ -69,7 +89,7 @@ en:
       update: Update Job Info
     title: Job Title
     view: View Job Posts
-    view_desc: This link will show you all job posts (old and new)
+    view_desc: This link will show you all job posts
```
Comment (lra-wtag) [inline config/locales/en.yml]: rewrite it as `This link will show you all job posts(old and new posts)`

File: config/locales/en.yml [hunk 14]
```diff
@@ -12,6 +12,26 @@ en:
     details: Candidate Details
     edit: Edit My Profile
     edu_profile: Educational Background
+    help_page:
+      address: House# 111, Road# 9, Block# F, Bashundhara R/A
+      cgpa: 2.4
+      contact: "+8801XXXXXXXXX"
+      degree: Bachelor of Computer Science and Engineering
```
Comment (lra-wtag) [inline config/locales/en.yml]: you can rewrite it as B.Sc in CSE

File: app/assets/stylesheets/main.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (lra-wtag) [reply app/assets/stylesheets/main.scss]: please google this pronto error

[the remark this answers is not in the document]
Comment (lra-wtag) [reply app/assets/stylesheets/main.scss]: please google this pronto error

File: app/views/pages/help.html.erb [hunk 15]
```diff
@@ -0,0 +1,88 @@
+<div class="candidate new">
+  <br/>
+<div class="form-group">
+  <%= link_to t('shared.all.back'), pages_path %>
+  <div class="container">
+    <h2 class="information-headline"><%= t('candidate.help_page.help') %></h2>
+    <h3><%= t('candidate.help_page.info') %></h3>
+
```
Comment (lra-wtag) [inline app/views/pages/help.html.erb]: no need of this empty line

File: app/assets/stylesheets/main.scss [hunk 16]
```diff
@@ -1057,6 +1067,17 @@ table tr td {
   overflow: hidden;
 }
 
+.help-link {
+  float: right;
+  height: 50px;
+  margin-right: 50px;
+  width: 50px;
```
Comment (lra-wtag) [inline app/assets/stylesheets/main.scss]: instead of using `px` use `rem`

File: config/routes.rb [hunk 17]
```diff
@@ -6,7 +6,11 @@
   root 'pages#index'
 
   resources :users
-  resources :pages
+  resources :pages, only: %i(index help) do
+    collection do
+      get 'help'
```
Comment (lra-wtag) [inline config/routes.rb]: try doing it in `show` method if possible

File: config/locales/en.yml [hunk 18]
```diff
@@ -12,6 +12,26 @@ en:
...
+      exp: 2
+      help: Candidate Info Help Page
+      hobby: Travel and sleep
+      info: This is help page to create your profile
+      key: Thinker
+      major: Data Mining
+      name: John Smith
+      pass_year: 2013
+      personal_interest: I dont know
+      plan: I want become a full stack developer
```
Comment (lra-wtag) [inline config/locales/en.yml]: rewrite it as `I want to become a Full Stack Backend Developer`

File: config/locales/en.yml [hunk 19]
```diff
@@ -12,6 +12,26 @@ en:
...
+      refer: Gavin Belson
+      skill: C, C++, Java, Ruby
```
Comment (lra-wtag) [inline config/locales/en.yml]: Rewrite it as: 
`Programming Languages: C, C++, Java, Ruby, Python`
`Framework: Spring Boot, JSP`
`Database: MySQL, Postgres`

File: app/assets/stylesheets/main.scss [hunk 20]
```diff
@@ -446,6 +449,11 @@ table tr td {
       margin-left: 10rem;
       width: 15rem;
     }
+
+    .gender-issue label {
```
Comment (sde-wtag) [inline app/assets/stylesheets/main.scss]: Please use proper scss syntax

File: app/views/pages/show.html.erb [hunk 21]
```diff
@@ -0,0 +1,87 @@
+<div class="candidate new">
+  <br/>
```
Comment (sde-wtag) [inline app/views/pages/show.html.erb]: Please do not use anu `<br />` tag inside html. If you really need any space here, use `css`

File: app/views/pages/show.html.erb [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+    <h3><%= t('candidate.help_page.info') %></h3>
+    <%= simple_form_for :help_page do |f| %>
+      <input id='step2' type='checkbox'>
+      <input id='step3' type='checkbox'>
+      <input id='step4' type='checkbox'>
+      <div id="part1" class="form-group">
+        <h3><%= t('candidate.basic_info') %></h3>
+          <%= f.input :name, label: t('candidate.name'), input_html: {value: t('candidate.help_page.name')}, readonly: true %>
+          <div class="gender-issue">
+            <%= f.label :gender %>
```
Comment (sde-wtag) [inline app/views/pages/show.html.erb]: As, you are using `simple_form`, you do not need to use `label` explicitely.

File: app/views/pages/show.html.erb [hunk 23]
```diff
@@ -0,0 +1,87 @@
+<div class="candidate new">
```
Comment (sde-wtag) [inline app/views/pages/show.html.erb]: Can we divide this `erb` file with some partial `view` file?

File: config/i18n-tasks.yml [hunk 24]
```diff
@@ -32,7 +32,7 @@ data:
   # This data is not considered unused and is never written to.
   external:
     ## Example (replace %#= with %=):
-    # - "<%#= %x[bundle show vagrant].chomp %>/templates/locales/%{locale}.yml"
+  # - "<%#= %x[bundle show vagrant].chomp %>/templates/locales/%{locale}.yml"
```
Comment (sde-wtag) [inline config/i18n-tasks.yml]: Unneccessay change 😞

File: config/routes.rb [hunk 25]
```diff
@@ -5,6 +5,8 @@
 
   root 'pages#index'
 
+  get 'pages/show', as: 'help'
```
Comment (sde-wtag) [inline config/routes.rb]: I am a bit confused about this route. In pages controller already you have show method, right? Or, am I missing something?

File: .gitignore [hunk 26]
```diff
@@ -25,3 +25,4 @@ public/system/
 spec/examples.txt
 
 .byebug_history
+.DS_Store
```
Comment (sde-wtag) [inline .gitignore]: No need to add this in `.gitignore`. You should be careful when you will push in github

File: app/views/helps/_additional.html.erb [hunk 27]
```diff
@@ -0,0 +1,5 @@
+<%= f.input :hobbies, label: t('shared.all.hobby'), input_html: {value: t('candidate.help_page.hobby')}, readonly: true%>
```
Comment (mmo-wtag) [inline app/views/helps/_additional.html.erb]: you have some spacing problem throughout your view files, like `{ <a space>put always an extra space before and after the curly braces<a space>}`, `<%= an extra space before and after in these erb tags %>`

File: spec/controllers/candidates_controller_spec.rb [hunk 28]
```diff
@@ -51,7 +50,6 @@
       expect(response).to render_template :show
     end
   end
-
```
Comment (mmo-wtag) [inline spec/controllers/candidates_controller_spec.rb]: please always keep an extra line between each `describe`, `context` and `it` blocks

File: app/controllers/helps_controller.rb [hunk 29]
```diff
@@ -0,0 +1,3 @@
+class HelpsController < ApplicationController
+  def show; end
```
Comment (mmo-wtag) [inline app/controllers/helps_controller.rb]: you've missed out to declare it in the `ability.rb` file. This file gives the permission to access stuffs. Please see other controllers and you'll get the idea about it. For more information either talk with Laila or have a look [here](https://github.com/CanCanCommunity/cancancan/wiki/Defining-Abilities) to get started with

**Pull request merged**